src/store_client.cc

```diff
@@ -506,6 +506,9 @@ store_client::readBody(const char *, ssize_t len)
     assert(_callback.pending());
     debugs(90, 3, "storeClientReadBody: len " << len << "");
 
+    if (len < 0)
+        return fail();
```

Comment:
IIRC, I looked at it and decided to leave it as is, possibly because I thought that `assert(_callback.pending())` above implies that `object_ok` cannot be false here, and I did not want to untangle this mess.

FWIW, I had to touch this code because dropped reads are now detected and manifested in negative lengths -- something this code mishandled before by calling `callback(len)`. SwapDir cannot return a successful read after a failed/dropped one, right? It if cannot, then it looks like we do not have to worry about `object_ok` in _this_ PR. Please correct me if you see a way for the object_ok to become important after this PR changes.

Reply:
> if you see a way for the object_ok to become important 

I have not found such reasons.